core/src/main/java/io/aiven/klaw/repository/TopicRequestsRepo.java

```diff
@@ -66,10 +66,10 @@ long countAllTopicRequestsByDescriptionAndTopictype(
   @Query(
       value =
           "select count(*) from kwtopicrequests where tenantid = :tenantId"
-              + " and teamid = :teamId and requestor != :requestor "
-              + "and topicstatus = :topicStatus  group by topicstatus",
+              + " and (teamid = :teamId or description = :teamId) and requestor != :requestor "
```

Comment:
that looks very strange...
who is the master of `description` field? can user somehow change it?
Why can't we keep up to date info about teamid in `teamid` field?

Reply:
Hey it is strange, it's a legacy issue with this request type, its definitely something that should be changed.
The teamId is the teamId of the requesting team and there is no additional field to place the teamId of the team that owns the topic, so it was placed in the description field.
Currently there is no way for the user to alter the description field on a claim type request.

For a bit of background a "Claim Request" is used when 1 team owns a topic and a second team wants to become the owner of that topic within Klaw.

Reply:
> so it was placed in the description field

is description is used only for this purpose? In case it is used for other purposes it could tend to different hidden bugs...

Reply:
On other request types the description can be manually entered by the user.

The bug that comes to mind would be if they entered a number that matched a teamId.

It is something that needs to be looked at. I mentioned this is a nice to have. if we have to wait to fix this it isnt going to block the release.

But its a bigger fix for sure.

Reply:
i would vote for fix it as a whole...
however another question that could pop up: should any migration be done? E.g. for already existing data

Reply:
I would think so i think we would have to cater for that